### PR TITLE
Java: Add DEL command

### PR DESCRIPTION
**Pull request by acarbonetto**
Adding the DEL() command (String command)

Example: 
```
// assuming key1 and key2 exist, while key "invalid" does not. 
Integer result = client.del(new String[] {"key1", "key2", "invalid"}).get();
assert result == 2;
```

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/StringCommands.java [hunk 1]
```diff
@@ -13,6 +13,15 @@
...
 public interface StringCommands {
 
+    /**
+     * Removes the specified <code>keys</code>. A key is ignored if it does not exist.
+     *
+     * @see <a href="https://redis.io/commands/del/">redis.io</a> for details.
+     * @param keys the keys we wanted to remove.
+     * @returns the number of keys that were removed.
+     */
+    CompletableFuture<Long> del(String[] keys);
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/StringCommands.java:23]: TODO: move to GenericBaseCommands

File: java/integTest/src/test/java/glide/TestUtilities.java [hunk 2]
```diff
@@ -14,11 +14,12 @@ public static BaseTransaction transactionTest(BaseTransaction baseTransaction) {
...
         baseTransaction.set(key2, "baz", SetOptions.builder().returnOldValue(true).build());
         baseTransaction.customCommand("MGET", key1, key2);
+        baseTransaction.del(new String[] {key1});
 
         return baseTransaction;
     }
 
     public static Object[] transactionTestResult() {
-        return new Object[] {"OK", null, new String[] {"bar", "baz"}};
+        return new Object[] {"OK", null, new String[] {"bar", "baz"}, 1L};
```
Comment (Elen-Ghulam) [inline java/integTest/src/test/java/glide/TestUtilities.java:23]: Can we replace "bar" with something else?

File: java/client/src/main/java/glide/api/commands/StringCommands.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/StringCommands.java:23]: next PR or this PR?

File: java/client/src/main/java/glide/api/commands/StringCommands.java [hunk 3]
```diff
@@ -13,6 +13,15 @@
  */
 public interface StringCommands {
 
+    /**
+     * Removes the specified <code>keys</code>. A key is ignored if it does not exist.
+     *
+     * @see <a href="https://redis.io/commands/del/">redis.io</a> for details.
+     * @param keys the keys we wanted to remove.
+     * @returns the number of keys that were removed.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StringCommands.java:21]:
```suggestion
     * Remove the specified <code>keys</code>. A key is ignored if it does not exist.
     *
     * @see <a href="https://redis.io/commands/del/">redis.io</a> for details.
     * @param keys The keys to remove.
     * @returns The number of keys that were removed.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 4]
```diff
@@ -114,6 +115,20 @@ public T info(InfoOptions options) {
         return getThis();
     }
 
+    /**
+     * Removes the specified <code>keys</code>. A key is ignored if it does not exist.
+     *
+     * @see <a href="https://redis.io/commands/del/">redis.io</a> for details.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:121]: please update javadoc here as well

File: java/client/src/main/java/glide/api/commands/StringCommands.java (hunk 3 above)
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/commands/StringCommands.java:21]: @return and not returns? @Yury-Fridlyand the Removes are from their code are you sure you want to change that?

File: java/client/src/test/java/glide/api/models/ClusterTransactionTests.java [hunk 5]
```diff
@@ -42,6 +43,9 @@ public void transaction_builds_protobuf_request() {
                                 .addArgs(RETURN_OLD_VALUE)
                                 .build()));
 
+        transaction.del(new String[] {"key1", "key2"});
```
Comment (SanHalacogluImproving) [inline java/client/src/test/java/glide/api/models/ClusterTransactionTests.java:46]: Why not add these tests to test utilities instead?

File: java/client/src/test/java/glide/api/models/TransactionTests.java [hunk 6]
```diff
@@ -41,6 +42,9 @@ public void transaction_builds_protobuf_request() {
                                 .addArgs(RETURN_OLD_VALUE)
                                 .build()));
 
+        transaction.del(new String[] {"key1", "key2"});
```
Comment (SanHalacogluImproving) [inline java/client/src/test/java/glide/api/models/TransactionTests.java:45]: Why not add these tests to test utilities instead?

File: java/integTest/src/test/java/glide/TestUtilities.java (hunk 2 above)
Comment (SanHalacogluImproving) [reply java/integTest/src/test/java/glide/TestUtilities.java:23]: That's what they have in their tests but I also recommend that we replace it with static value variables as we are going to implement multiple commands here.  (have an upstream PR for it) https://github.com/aws/glide-for-redis/pull/955/files#diff-2e2a26bd6d68b97dc3ec663ffa8f00287190a7107235b4caacf547ddd606c419R10

File: java/client/src/main/java/glide/api/commands/StringCommands.java (hunk 3 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/StringCommands.java:21]: Oh right, please use `@return` in java, not `@returns`

File: java/client/src/main/java/glide/api/commands/StringCommands.java (hunk 1 above)
Comment (acarbonetto) [reply java/client/src/main/java/glide/api/commands/StringCommands.java:23]: This PR... just getting to it

File: java/integTest/src/test/java/glide/TestUtilities.java (hunk 2 above)
Comment (acarbonetto) [reply java/integTest/src/test/java/glide/TestUtilities.java:23]: Sorry - I just copied from their tests without thinking.  
I switched to apples/oranges in the examples app, and we can do that here too.

File: java/client/src/test/java/glide/api/models/TransactionTests.java (hunk 6 above)
Comment (acarbonetto) [reply java/client/src/test/java/glide/api/models/TransactionTests.java:45]: That's for IT tests. This is for UT tests.  
That being said... we could probably create a BaseTransactionTests.java file that does something similar.

File: java/client/src/main/java/glide/api/commands/StringCommands.java (hunk 3 above)
Comment (acarbonetto) [reply java/client/src/main/java/glide/api/commands/StringCommands.java:21]: Changing documentation should be a separate PR and we should update all clients. 
This is from the Node documentation: https://github.com/aws/glide-for-redis/blob/main/node/src/BaseClient.ts#L418

File: java/client/src/test/java/glide/api/models/ClusterTransactionTests.java (hunk 5 above)
Comment (acarbonetto) [reply java/client/src/test/java/glide/api/models/ClusterTransactionTests.java:46]: That's for IT tests. This is for UT tests.

**Review by Yury-Fridlyand — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ * @see <a href="https://redis.io/commands/?group=generic">Generic Commands</a>
+ */
+public interface GenericBaseCommands {
+
+    /**
+     * Removes the specified <code>keys</code>. A key is ignored if it does not exist.
+     *
+     * @see <a href="https://redis.io/commands/del/">redis.io</a> for details.
+     * @param keys the keys we wanted to remove.
+     * @returns the number of keys that were removed.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:18]:
```suggestion
     * @return the number of keys that were removed.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 8]
```diff
@@ -114,6 +115,20 @@ public T info(InfoOptions options) {
...
+     * @param keys the keys we wanted to remove.
+     * @returns the number of keys that were removed.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:123]:
```suggestion
     * @return The number of keys that were removed.
```

File: java/client/src/test/java/glide/api/models/ClusterTransactionTests.java (hunk 5 above)
Comment (acarbonetto) [reply java/client/src/test/java/glide/api/models/ClusterTransactionTests.java:46]: See: https://github.com/Bit-Quill/glide-for-redis/pull/91

**Pull request merged**